pycqed/instrument_drivers/physical_instruments/QuTech_QCC.py

```diff
@@ -0,0 +1,615 @@
+"""
+INT32_MAX = +2147483647
+INT32_MIN = -2147483648
+CHAR_MAX = +127
+CHAR_MIN = -128
+
+MAX_NUM_INSN = 2**15
+
+class QCC(SCPI):
+    """
+    This is class is used to serve as the driver between the user and the
+    QCC hardware. The class starts by querying the hardware via the
+    SCPI interface. The hardware then responds by providing the available
+    standard qcodes parameters. This class then uses qcodes to automatically
+    generate the functions necessary for the user to control the hardware.
+    """
+    exceptionLevel = logging.CRITICAL
+
+    def __init__(self, name, address, port, log_level=False, **kwargs):
+        self.model = name
+        self._dummy_instr = False
+        self.driver_version = "0.1.0"
+        try:
+            super().__init__(name, address, port, **kwargs)
+        except Exception as e:
+            # Setting up the SCPI sometimes fails the first time.  If this
+            # happens a second effort to initialize and settup the connection
+            # is made
+            print("Failed to connect (" + str(e) + "). The system will retry" +
+                  " to connect")
+            self.remove_instance(self)
+            super().__init__(name, address, port, **kwargs)
+        self.get_idn()
+        self.add_standard_parameters()
+        self.add_additional_parameters()
+        self._init_submodules()
+        self.connect_message()
+
+    def _init_submodules(self):
+        """
+        The parser helper objects are initialized in this function.
+        """
+        self.microcode = QCCMicrocode()
+        self.QISA = QISA_Driver()
+
+        """
+        QCC only works with version 4.0.0 of the assembler
+        """
+        if self.QISA.getVersion() != '4.0.0':
+            sys.exit('The QISA Assembler installed in the environment does not match version 4.0.0, the only supported for running QCC.')
+
+        curdir = os.path.dirname(__file__)
+        qmap_fn = os.path.join(curdir, '_QCC', 'qisa_opcodes.qmap')
+
+        """
+        Assembler now aditionally requires quantum layout information file
+        """
+        configureinput = os.path.join(curdir, '_QCC', 'quantum_layout_information_17.txt')
+        if not os.path.isfile(configureinput):
+        	sys.exit('The QISA Assembler supporting QCC now expects a quantum_layout_information file in the Pycqed physical instruments directory.')
+
+        self.QISA.read(configureinput)
+
+        self.QISA.enableScannerTracing(False)
+        self.QISA.enableParserTracing(False)
+        self.QISA.setVerbose(False)
+
+        self.qisa_opcode(qmap_fn)
+
+    def stop(self, getOperationComplete=True):
+        self.run(0),
+        self.enable(0)
+        # Introduced to work around AWG8 triggering issue
+        if getOperationComplete:
+            self.getOperationComplete()
+
+    def start(self, getOperationComplete=True):
+        self.enable(1)
+        self.run(1)
+        # Introduced to work around AWG8 triggering issue
```

Comment:
Same here..

Reply:
I have no indication otherwise. Therefore, maintaining this as part of driver